hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/ReversedStoreScanner.java

```diff
@@ -131,6 +134,9 @@ public boolean backwardSeek(Cell key) throws IOException {
     if (checkFlushed()) {
       reopenAfterFlush();
     }
+    if (checkIfScannersToBeReset()) {
```

Comment:
Like flush you also check for Compaction? Shoudl it be done under one method only ?

Reply:
For flush, we need to do `reopenAfterFlush()` and for compaction, now we need to do `reopenScanners()`, hence kept it separately. 
You are asking for something like this?
```
if(checkFlushed() || checkIfScannersToBeReset()){
   // perform common actions
}
```

Reply:
Since we have separate volatile booleans for flush and compaction, so thought of having separate methods to check for this compaction related condition. Should be fine?